File: lingji-gateway/handler/routing.go
package handler

import (
	"strings"

	"github.com/AUrlius/lingji-gateway/protocol"
)

const DefaultAgentID = "lingji-pc"

// IsAgentDevice 判断 device_id 是否为 PC Agent（lingji-* 前缀，非 phone-*）
func IsAgentDevice(deviceID string) bool {
	return strings.HasPrefix(deviceID, "lingji-")
}

// resolveTargetAgentID 从上行消息 payload 解析目标 Agent，缺省为 lingji-pc
func resolveTargetAgentID(raw []byte) string {
	msg, err := protocol.ParseMessage(string(raw))
	if err != nil {
		return DefaultAgentID
	}
	if id, ok := msg.Payload["target_agent_id"].(string); ok && id != "" {
		return id
	}
	return DefaultAgentID
}

File: lingji-gateway-node/src/handler/routing.js
const DEFAULT_AGENT_ID = 'lingji-pc';

export function isAgentDevice(deviceId) {
  return typeof deviceId === 'string' && deviceId.startsWith('lingji-');
}

/** @param {string} raw */
export function resolveTargetAgentID(raw) {
  try {
    const msg = JSON.parse(raw);
    const id = msg.payload?.target_agent_id;
    if (typeof id === 'string' && id) return id;
  } catch {
    /* ignore */
  }
  return DEFAULT_AGENT_ID;
}

export { DEFAULT_AGENT_ID };
